src/example/prompt.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable()
+export class PullRequestReviewService {
+  constructor(private readonly httpService: HttpService) {}
+
+  async getPullRequestDetails(repo: string, prId: number): Promise<any> {
+    const url = `https://api.github.com/repos/${repo}/pulls/${prId}`;
+    return this.httpService.get(url).pipe(map(response => response.data)).toPromise();
+  }
+
+  async reviewPullRequest(repo: string, prId: number): Promise<void> {
+    const details = await this.getPullRequestDetails(repo, prId);
+
+    if (details.changed_files > 10) {
+      this.addComment(repo, prId, "Too many files changed in this PR.");
+    }
+
+    if (!details.title.includes("JIRA-")) {
+      this.addComment(repo, prId, "PR title does not include JIRA ticket.");
+    }
+
+    this.mergePullRequest(repo, prId);
+  }
```

Comment:
PullProofAI

The `reviewPullRequest` method seems to be doing multiple tasks (checking file changes, title, and merging). Consider refactoring it into smaller, more focused methods to adhere to the single responsibility principle.
```diff
+   async checkFileChanges(repo: string, prId: number): Promise<void> {
+       const details = await this.getPullRequestDetails(repo, prId);
+       if (details.changed_files > 10) {
+           this.addComment(repo, prId, "Too many files changed in this PR.");
+       }
+   }
+
+   async checkTitle(repo: string, prId: number): Promise<void> {
+       const details = await this.getPullRequestDetails(repo, prId);
+       if (!details.title.includes("JIRA-")) {
+           this.addComment(repo, prId, "PR title does not include JIRA ticket.");
+       }
+   }
+
+   async mergeIfReady(repo: string, prId: number): Promise<void> {
+       this.mergePullRequest(repo, prId);
+   }
+
+   async reviewPullRequest(repo: string, prId: number): Promise<void> {
+       await this.checkFileChanges(repo, prId);
+       await this.checkTitle(repo, prId);
+       await this.mergeIfReady(repo, prId);
+   }
```



<!-- This is an auto-generated comment by PullProofAI -->